### PR TITLE
Better names and details for build operations

### PR DESCRIPTION
**Pull request by oehme**
@alkemist as discussed on our call, I have 

- given dedicated details to all build operations
- linked the result type with the details type
- renamed the classes that we thought were badly named

Since I'm going on vacation soon, I'd like to hand this over to you. I can still do some polishing tomorrow though.

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/execution/taskgraph/CalculateTaskGraphDescriptor.java [hunk 1]
```diff
@@ -17,17 +17,20 @@
...
 /**
- * Operation Descriptor for Calculating the task graph of a build.
+ * Details about the task graph of a build.
+ *
  * This class is intentionally internal and consumed by the build scan plugin.
  *
  * @since 4.0
  */
-public class CalculateTaskGraphDescriptor {
+public final class CalculateTaskGraphDescriptor implements BuildOperationDetails<CalculateTaskGraphOperationResult> {
```
Comment (breskeby) [inline subprojects/core/src/main/java/org/gradle/execution/taskgraph/CalculateTaskGraphDescriptor.java]: We should probably rename this to CalculateTaskGraphDetails? I can make this as part of my story...

Comment (alkemist) [inline subprojects/core/src/main/java/org/gradle/execution/taskgraph/CalculateTaskGraphDescriptor.java]: I've done this.

**Comment by alkemist** [comment]
I've made all of the changes that I want to.

**Comment by breskeby** [comment]
are we good to go here? I can take care on merging.

**Comment by oehme** [comment]
Commit build succeeded on the branch, so yes, this is good to go.

**Comment by oehme** [comment]
Note that this will probably break the Gradleception build, because we use the build scan plugin 1.7, which relies on the old names of the types we changed here. So we might wanna remove that until there is a fixed plugin version.

**Comment by breskeby** [comment]
@oehme I've disabled build scans for now for the gradleception build

**Comment by big-guy** [comment]
Is it OK to break the build scan plugin like this?

**Comment by breskeby** [comment]
yes, we discussed this. 1.7 is already broken for gradle 4.0 snapshots and I’ll work on adjusting the build scan plugin.

**Pull request merged**